### ISSUE TITLE
W4A4 Quantization: Cannot Reproduce ScienceQA Results

### ISSUE THREAD

**Issue by zhongyi-nku** [SAI-Lab-NYU/QSVD#2]
Thanks so much for open-sourcing QSVD — it's a great framework!

I followed the docs to run ScienceQA evaluation with the **default W4A4 quantization config** (no custom parameter changes), but my results are noticeably lower than the numbers in your paper.

Could you share a few key details to help reproduce the paper's results? For example:
1. Did you use specific calibration data/sample counts for W4A4?
2. Any task-specific tweaks for cross-layer rank allocation?
3. Prompt template/inference params for ScienceQA?

Thanks again for your help!

**Comment by ytwang13** [comment]
Thanks for your interests!
1. we use scienceQA training data and sample size of 256 
2. no tweaks for rank allocation
3. the SQA evaluation details can be found in https://github.com/SAI-Lab-NYU/QSVD/blob/95615cf316fc5554920b079e81feb20d838857e7/fake_quant/eval_utilsdistllava.py#L197
4. the prompt templet can be found in message_to_prompt function https://github.com/SAI-Lab-NYU/QSVD/blob/95615cf316fc5554920b079e81feb20d838857e7/fake_quant/eval_utilsdistllava.py#L63

And can you share the environment/script you are using or  did you try different random seeds?

**Comment by zhongyi-nku** [comment]
Hi,

Thanks for your reply!

I ran W4A4 quantization/evaluation with this script (repo files unmodified, matched ScienceQA Train + 256 samples as you mentioned):
```bash
cd QSVD/fake_quant
python mainllava.py --model liuhaotian/llava-v1.5-7b  \
                --a_bits 4 --w_bits 4 \
                --cal_dataset ScienceQA_Train --eval_dataset ScienceQA_TEST \
                --nsamples 256 --seed 0 --svd_mode "U" \
                --rank_ratio 1.5 --had_rank --svd_lm \
                --vit_module --rotate --beta_then_svd \
                --w_rtn --w_clip --lma_clip_ratio 0.9 --qkv_fuse \
                --calib_method 'abs_mean' --act_aware --act_alpha 0.5 \
                --vit_online --grad_info --setting "/sqa/online_then_qkvlm_svd/seed0"
```

I checked `eval_utilsdistllava.py` (Line 197 `generate` / Line 63 `message_to_prompt`) and used **raw ScienceQA (no preprocessing)**.

Quick questions:
1. Did you perform any preprocessing on the ScienceQA dataset (e.g., formatting/filtering) before running the script?
2. You mentioned "no tweaks for rank allocation" – do I need to remove/tweak `--rank_ratio 1.5/--had_rank/--svd_lm` in my command?
3. Is my full parameter setup correct for W4A4 on ScienceQA? (Env: PyTorch 2.0.1 + CUDA 11.8, only tested seed=0)

Thanks!

**Comment by ytwang13** [comment]
1. yes, we process the ScienceQA based on the llava repo, and we keep only the img part like provided in https://github.com/SAI-Lab-NYU/QSVD/blob/95615cf316fc5554920b079e81feb20d838857e7/myllava/scripts/convert_sqa_to_llava.py#L24
2. the command is correct, you can try different random seed
3. we haven't test the code in pytoch 2.0.1 maybe follow the installation using pytorch 2.6.0 https://github.com/SAI-Lab-NYU/QSVD/blob/95615cf316fc5554920b079e81feb20d838857e7/README.md?plain=1#L41
